### PR TITLE
Feature/sg 111 kd teacher input adaptor

### PR DESCRIPTION
**Pull request by shaydeci**
Decinet training uses teacher with different normalization values, hence we need to add support for adding an “input adaptor” that takes the student input and transforms it to the form the teacher is successful at.

### CODE REVIEW

File: tests/unit_tests/kd_model_test.py [hunk 1]
```diff
@@ -205,6 +206,21 @@ def test_train_kd_module_pretrained_ckpt(self):
...
+        kd_model = KDModel("build_kd_module_with_with_input_adapter", device='cpu')
+        kd_model.build_model(student_architecture='resnet18', teacher_architecture='resnet50',
+                             student_arch_params={'num_classes': 1000}, teacher_arch_params={'num_classes': 1000},
+                             checkpoint_params={'teacher_pretrained_weights': "imagenet"},
+                             arch_params={
+                                 "teacher_input_adapter": NormalizationAdapter(mean_original=[0.485, 0.456, 0.406],
+                                                                               std_original=[0.229, 0.224, 0.225],
+                                                                               mean_required=[0.5, 0.5, 0.5],
+                                                                               std_required=[0.5, 0.5, 0.5])})
+        imagenet_resnet50_sg_model = SgModel("pretrained_resnet50")
```
Comment (lotem-deci) [inline tests/unit_tests/kd_model_test.py:219]: @shaydeci  - What are lines 219-221 for?

Comment (shaydeci) [reply tests/unit_tests/kd_model_test.py]: fixed

**Review by ofrimasad — COMMENTED** [review]
minor comment. just add the documentation and other than that LGTM

File: src/super_gradients/training/utils/module_utils.py [hunk 2]
```diff
@@ -194,3 +194,24 @@ def __init__(self,
...
         return self.seq(x)
+
+
+class NormalizationAdapter(torch.nn.Module):
+    """
+    Denormalizes input by mean_original, std_original, then normalizes by mean_required, std_required.
+
+    Used in KD training where teacher expects data normalized by mean_required, std_required.
+    """
+    def __init__(self, mean_original, std_original, mean_required, std_required):
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/utils/module_utils.py:209]: added

**Review by ofrimasad — APPROVED** [review]
LGTM
